bin/ajax

```diff
@@ -78,63 +76,84 @@ eb_can_clean_ceph = 'eb0.xenon.local'
-
-            log.info(f'Done with {args.clean}')
-        elif args.clean == 'all':
-            while True:
-                if hostname == eb_can_clean_ceph:
-                    clean_ceph()
-                    clean_abandoned(delete_live=True)
-                    # clean_database(delete_live=True)
-
-                    # Don't do clean_unregistered by default (only if specified
-                    # as specific argument on host "eb_can_clean_ceph") as
-                    # someone might be storing some test data in the /live_data
-                    # folder
-                    # clean_unregistered(delete_live=True)
-
-                # Not yet implemented functions
-                # clean_high_level_data()
-                # clean_raw_records_nv()
-
-                clean_abandoned()
-                clean_non_latest()
-
-                # These two modes below shouldn't be done on autopilot at the
-                # time of writing.
-                # clean_unregistered()
-                # clean_database()
-                if not args.execute:
-                    break
-                log.info(f'Loop finished, take a {ajax_thresholds["nap_time"]} s nap')
-                time.sleep(ajax_thresholds['nap_time'])
-        else:
-            raise ValueError(f'Unknown cleaning mode {args.clean}')
+        remove_run_from_host(args.number, delete_live=args.delete_live,
+                             force=args.force, reason='manual')
+
+    if not args.clean:
+        raise ValueError('either --number or --clean should be specified')
+
+    _modes = {'ceph': clean_ceph,
+              'high_level_data': clean_high_level_data,
+              'non_latest': clean_non_latest,
+              'unregistered': clean_unregistered,
+              'raw_records_nv': clean_raw_records_nv,
+              'abandoned': clean_abandoned,
+              'old_hash': clean_old_hash,
+              'database': clean_database}
+
+    if args.clean not in _modes and args.clean != 'all':
+        raise ValueError(f'Unknown cleaning mode {args.clean}')
+
+    if args.clean != 'all':
+        # Do this mode
+        _modes[args.clean]()
+        if hostname == eb_can_clean_ceph and (args.clean in
+                                              ['abandoned', 'database']):
+            _modes[args.clean](delete_live=True)
+        log.info(f'Done with {args.clean}')
+        return
+
+    # Apparently we want to clean everything. Let's loop:
+    while True:
+        try:
+            if hostname == eb_can_clean_ceph:
+                clean_ceph()
+                clean_abandoned(delete_live=True)
+                # clean_database(delete_live=True)
+
+                # Don't do clean_unregistered by default (only if specified
+                # as specific argument on host "eb_can_clean_ceph") as
+                # someone might be storing some test data in the /live_data
+                # folder. That is bad practise though

```

Comment:
And for this reason, people should not complain if they store test data _in the production folder_ and it gets deleted. Ceph has many folders for storing test data. `/live_data/xenonnt` is not one of them.